app/src/main/java/de/xikolo/utils/LanguageUtil.kt

```diff
@@ -1,18 +1,34 @@
 package de.xikolo.utils
 
+import android.content.Context
 import java.util.Locale
 
 object LanguageUtil {
+    var deviceLanguage: String = Locale.getDefault().language.takeIf { it != "zh" } ?: "cn"
+
     @OptIn(ExperimentalStdlibApi::class)
     fun toNativeName(tag: String): String {
-        val languageTag =
-            if (tag == "cn") { // work-around for bug in the backend
-                "zh"
-            } else {
-                tag
-            }
+        val languageTag = correctLanguageTag(tag)
 
         val locale = Locale.forLanguageTag(languageTag)
         return locale.getDisplayLanguage(locale).capitalize(locale)
     }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    fun toLocaleName(context: Context, tag: String): String {
+        val languageTag = correctLanguageTag(tag)
+        val locale = Locale.forLanguageTag(languageTag)
+        val deviceLocale = Locale(Locale.getDefault().language)
```

Comment:
Is this necessary, wouldn't `Locale.getDefault()` suffice?

Reply:
Yes, it suffices :) good catch!